src/test/java/seedu/address/testutil/SetPersonDescriptorBuilder.java

```diff
@@ -0,0 +1,130 @@
+        this.descriptor = new SetPersonDescriptor();
+        this.descriptor.setName(person.getName());
+        this.descriptor.setAddress(person.getAddress());
+        this.descriptor.setRole(person.getRole());
+        this.descriptor.setTimezone(person.getTimezone());
+        this.descriptor.setTags(person.getTags());
+
+        for (ContactType key : person.getContacts().keySet()) {
+            this.descriptor.setContacts(key, person.getContacts().get(key));
+        }
+    }
+
+    /**
+     * Sets the name of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withName(String name) {
+        descriptor.setName(new Name(name));
+        return this;
+    }
+
+    /**
+     * Sets the address of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withAddress(String address) {
+        descriptor.setAddress(new Address(address));
+        return this;
+    }
+
+    /**
+     * Sets the role of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withRole(String role) {
+        descriptor.setRole(new Role(role));
+        return this;
+    }
+
+    /**
+     * Sets the timezone of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withTimezone(String timezone) {
+        descriptor.setTimezone(new Timezone(timezone));
+        return this;
+    }
+
+    /**
+     * Sets the tags of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withTags(String... tags) {
+        Set<Tag> tagSet = Stream.of(tags).map(Tag::new).collect(Collectors.toSet());
+        descriptor.setTags(tagSet);
+        return this;
+    }
+
+    /**
+     * Sets the telegram Contact of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withTelegram(String telegram) {
+        Telegram newTelegram = new Telegram(telegram);
+        this.descriptor.setContacts(ContactType.TELEGRAM, newTelegram);
+        return this;
+    }
+
+    /**
+     * Sets the email Contact of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withEmail(String email) {
+        Email newEmail = new Email(email);
+        this.descriptor.setContacts(ContactType.EMAIL, newEmail);
+        return this;
+    }
+
+    /**
+     * Sets the phone Contact of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withPhone(String phone) {
+        Phone newPhone = new Phone(phone);
+        this.descriptor.setContacts(ContactType.PHONE, newPhone);
+        return this;
+    }
+
+    /**
+     * Sets the slack Contact of the SetPersonDescriptor we are building.
+     */
+    public SetPersonDescriptorBuilder withSlack(String slack) {
+        Slack newSlack = new Slack(slack);
+        this.descriptor.setContacts(ContactType.SLACK, newSlack);
+        return this;
+    }
```

Comment:
```
public SetPersonDescriptorBuilder withContact(ContactType type, String value) {
    this.descriptor.setContacts(type, value)
}
```
is more extensible.

Reply:
`SetPersonDescriptor::setContact` takes in a `Contact`, not a String so I might need to use a switch-case there. Is that still preferable over this?

Reply:
I see. Thanks for the tip!